diesel_codegen/tests/associations.rs

```diff
@@ -0,0 +1,149 @@
+#![allow(dead_code, unused_must_use)]
+
+#[cfg(feature = "sqlite")]
+type Backend = ::diesel::sqlite::Sqlite;
+#[cfg(feature = "mysql")]
+type Backend = ::diesel::mysql::Mysql;
+#[cfg(feature = "postgres")]
+type Backend = ::diesel::pg::Pg;
+
+#[test]
+fn simple_belongs_to() {
+    use diesel::*;
+
+    table! {
+        users {
+            id -> Integer,
+            name -> Text,
+        }
+    }
+
+    table! {
+        posts {
+            id -> Integer,
+            user_id -> Integer,
+            title -> Text,
+        }
+    }
+
+    #[derive(Identifiable)]
+    pub struct User {
+        id: i32,
+        name: String
+    }
+
+    #[derive(Associations, Identifiable)]
+    #[belongs_to(User)]
+    pub struct Post {
+        id: i32,
+        user_id: i32,
+        title: String,
+    }
+
+
+    posts::table.inner_join(users::table)
+        .select((users::id, users::name))
+        .filter(posts::id.eq(1)
+            .and(posts::user_id.eq(2))
+                .and(posts::title.eq("Bar")));
```

Comment:
Can you name these "compile-success assertions" by making them let statements? E.g., `let _can_select_columns_from_join = posts::table.inn…` (or a better name)